mfr/core/extension.py

```diff
@@ -15,38 +29,90 @@ def __init__(self, ext, source_file_path, output_file_path, format):
         :param format: the format of the exported file (e.g. 1200*1200.jpg)
         """
 
-        self.ext = ext
-        self.source_file_path = source_file_path
-        self.output_file_path = output_file_path
+        self.metadata = metadata
+        self.ext = metadata.ext
+        self.input_stream = input_stream
         self.format = format
+        self.source_file_id = uuid.uuid4()
+        self.cache_file_id = '{}.{}'.format(self.metadata.unique_key, self.format)
+        self.local_cache_provider = waterbutler.core.utils.make_provider(
+            'filesystem', {}, {}, settings.LOCAL_CACHE_PROVIDER_SETTINGS
+        )
+        self.exporter_name = utils.get_exporter_name(self.metadata.ext)
+        if self.exporter_name:
+            cache_file_path_str = '/export/{}.{}'.format(self.cache_file_id, self.exporter_name)
+        else:
+            cache_file_path_str = '/export/{}'.format(self.cache_file_id)
+
         self.exporter_metrics = MetricsRecord('exporter')
         if self._get_module_name():
             self.metrics = self.exporter_metrics.new_subrecord(self._get_module_name())
 
+    async def __call__(self):
+
+        self.source_wb_path = await self.local_cache_provider.validate_path(
+            '/export/{}'.format(self.source_file_id)
+        )
+        self.source_file_path = self.source_wb_path.full_path
+        self.output_file_id = '{}.{}'.format(self.source_wb_path.name, self.format)
+        self.output_wb_path = await self.local_cache_provider.validate_path(
+            '/export/{}'.format(self.output_file_id)
+        )
+        self.output_file_path = self.output_wb_path.full_path
         self.exporter_metrics.merge({
```

Comment:
I need to know more about what these metrics are trying to cover so I make sure this does'nt break data we've already collected.